pkg/handlers/query.go

```diff
@@ -1,22 +1,68 @@
 package handlers
 
 import (
-	"fmt"
+	"database/sql"
+	"encoding/json"
 	"net/http"
+	"reflect"
 
 	gabi "github.com/app-sre/gabi/pkg"
 )
 
+type QueryData struct {
+	Query string
+}
+
 func Query(env *gabi.Env) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		// https://stackoverflow.com/a/46021789
 		env.Logger.Info("testing")
-		fmt.Fprintf(w, "%s %s %s \n", r.Method, r.URL, r.Proto)
+		// fmt.Fprintf(w, "%s %s %s \n", r.Method, r.URL, r.Proto)
 		for k, v := range r.Header {
-			fmt.Fprintf(w, "Header field %q, Value %q\n", k, v)
+			// fmt.Fprintf(w, "Header field %q, Value %q\n", k, v)
 			env.Logger.Infof("Header field %q, Value %q", k, v)
 		}
-		fmt.Fprintf(w, "Host = %q\n", r.Host)
-		fmt.Fprintf(w, "RemoteAddr= %q\n", r.RemoteAddr)
+		// fmt.Fprintf(w, "Host = %q\n", r.Host)
+		// fmt.Fprintf(w, "RemoteAddr= %q\n", r.RemoteAddr)
+
+		var q QueryData
+		err := json.NewDecoder(r.Body).Decode(&q)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		rows, err := env.DB.Query(q.Query)
```

Comment:
We need to handle a missing database table here (I think). Try testing this with two queries against the test RDS instance:

`SELECT * FROM actor;` - working query with existing table
`SELECT * FROM companies;` - broken query, results in panic. doesn't crash the program, and the server continues to respond to incoming requests.

It may be OK to panic, but from the requester's perspective, they get a client connection closed error with no feedback on what the problem with their query was. Ideally, we can catch this error and provide helpful feedback to the user instead of forcibly closing the connection.